### ISSUE TITLE
REST API for user management

### ISSUE THREAD

**Issue by marhew** [bptlab/chimera#75]
To implement user management capabilities (see #9) we need to implement a REST API that contains endpoints to create, read, update, delete users and organizations, as well as assign users to organizations and case models to organizations.

**Resources**
We have 2 main resources: **organizations** and **users**. 
```
GET | POST  /organizations
GET | POST  /users
```
We can GET, DELETE, and change (PUT) users and organizations
```
GET | PUT | DELETE  /organizations/123
GET | PUT | DELETE  /users/123
```
Organizations have organizational roles that can be requested and created. Also they have case models that belong to the organization. The response should contain links to the case models. Organizations also have owners and members that can be added or requested. Both are users, so the response should link them.
```
GET | POST  /organizations/123/roles
GET | POST  /organizations/123/casemodels
GET | POST  /organizations/123/owners
GET | POST  /organizations/123/members
```
Members can be deleted (without deleting the user), meaning that the user is no longer member of the organization. Members of organizations can have several roles. Roles of members need to be roles of the organization. It is possible to grant roles to members (`POST`) and revoke them (`DELETE`).
Members can have several cases they participate in, the endpoint can be used to realize a "my cases" view in the frontend. Cases of a member need to be linked to the case (`/casemodel/xyz/case/qwe`) so that the frontend can query and display all cases of a logged in user.
```
GET | DELETE  /organizations/123/members/789
GET | POST  /organizations/123/members/789/roles
GET | POST  /organizations/123/members/789/cases
GET | DELETE /organizations/123/members/789/roles/22
```
**Authorization / visibility**
When requesting the list of organizations (`GET /organizations`) the response will contain **all** organizations regardless whether the currently logged in user (as identified by the authentication header) is a member or not. However, when requesting a specific organization `GET /organizations/123`) the API checks whether the currently logged in user is a member of that organization. If this is not the case, the response will be *"You are not a member of this organization, and cannot view organizational details"*.

Similarly, all members of an organization can request the list of case models (`GET /organizations/123/casemodels`). The response should contain the roles that are allowed to view the case model and/or start cases. Only if the logged in user has the specified role they can request a specific case model (`GET /organizations/123/casemodels/789`). 

**Open Questions**
- We need to define the data structure of the resources. What information does an organization / a user contain? How to we model it?
- How to represent that visibility of case models in organizations it restricted by organizational roles? `/organizations/123/casemodels/456/allowedroles`?

**Acceptance Criteria**
- the API needs to be documented in swagger. This should happen automatically #100 
- the Java code implementing the endpoints is documented

**Comment by marhew** [comment]
Please consider the discussion in #98

**Comment by marhew** [comment]
**Assigning members**: So far, no invitation system. Owners of an organization can directly add members to the organization.

**Comment by janwestphal306** [comment]
What information should be seen if we request a specific user by `GET /users/123` and who should be able to see it. Should these information be used for some kind of profile?

**Comment by janwestphal306** [comment]
What about the deletion of `casemodels`, `owners` and `roles` of an organization?

**Comment by marhew** [comment]
The last `owner` of an organization can not be deleted. Other owners that are deleted still stay members of the organization. 
I will think about the other two

**Comment by janwestphal306** [comment]
Which information should be in the response for `GET /organizations/123/members/789`? Information like roles?

**Comment by janwestphal306** [comment]
Implemented with PR #143
